### DISCUSSION TITLE
proxy2.1.2的collection命名问题

### DISCUSSION THREAD

**Discussion by bingbb47** [milvus-io/milvus#19273] [Q&A and General discussion] [answered]
你好，
关于collection的命名，在之前的版本是支持'_'与'$'两种字符的的
当我升级了2.1.2版本之后，以$为分隔符创建失败了，返回（Invalid collection name: new$test. Collection name can only contain numbers, letters and underscores）
请问这是属于bug还是新版本的修改呢
另，
我发现使用2.1.1版本的proxy时不会有此限制，
那么我是否将2.1.1版本的proxy搭配2.1.2版本的其他组件一起使用呢？

谢谢

**Comment by bingbb47** [comment, marked as answer]
> 

刚看了下代码，确实是取消了$，可以介绍一下为什么取消么

**Reply by yhmo** [reply]
因为内部在匹配partition name的时候用到了正则表达式，而$又是正则表达式的特殊符号，所以导致名字里带有$的partiiton都无法被匹配到。所以我们把collection/partition/field/index name都限制在英文字符，数字以及下划线的范围内。

**Reply by bingbb47** [reply]
那对于collection有没有可能恢复$的使用呢？在一些场景中需要一些特殊字符作为分隔符使用，现阶段仅有'_'

**Reply by yhmo** [reply]
从维护的角度来看，命名规则还是简单点好，并且规则统一，这样以避免可能潜在的bug风险。如果以后有很多用户反映说命名规则太僵化，那还是有可能会修改代码以支持某些特殊符号的。
